File: android-app/app/src/main/java/com/aiproj/mobile/data/models/Task.kt
package com.aiproj.mobile.data.models

import com.google.gson.annotations.SerializedName

/**
 * 任务数据模型
 */
data class Task(
    @SerializedName("id")
    val id: Int,

    @SerializedName("project_id")
    val projectId: Int,

    @SerializedName("title")
    val title: String,

    @SerializedName("description")
    val description: String?,

    @SerializedName("status")
    val status: TaskStatus,

    @SerializedName("priority")
    val priority: TaskPriority?,

    @SerializedName("assignee_id")
    val assigneeId: Int?,

    @SerializedName("assignee")
    val assignee: User?,

    @SerializedName("parent_id")
    val parentId: Int?,

    @SerializedName("due_date")
    val dueDate: String?,

    @SerializedName("estimated_minutes")
    val estimatedMinutes: Int?,

    @SerializedName("actual_minutes")
    val actualMinutes: Int?,

    @SerializedName("children_count")
    val childrenCount: Int?,

    @SerializedName("created_at")
    val createdAt: String,

    @SerializedName("updated_at")
    val updatedAt: String
)

/**
 * 任务状态枚举
 */
enum class TaskStatus {
    @SerializedName("draft")
    DRAFT,

    @SerializedName("planning")
    PLANNING,

    @SerializedName("todo")
    TODO,

    @SerializedName("in_progress")
    IN_PROGRESS,

    @SerializedName("testing")
    TESTING,

    @SerializedName("completed")
    COMPLETED,

    @SerializedName("cancelled")
    CANCELLED,

    @SerializedName("on_hold")
    ON_HOLD,

    @SerializedName("blocked")
    BLOCKED,

    @SerializedName("archived")
    ARCHIVED
}

/**
 * 任务优先级枚举
 */
enum class TaskPriority {
    @SerializedName("low")
    LOW,

    @SerializedName("medium")
    MEDIUM,

    @SerializedName("high")
    HIGH
}

/**
 * 任务列表响应
 */
data class TaskListResponse(
    @SerializedName("tasks")
    val tasks: List<Task>,

    @SerializedName("total")
    val total: Int,

    @SerializedName("pagination")
    val pagination: Pagination?
)

/**
 * 创建/更新任务请求
 */
data class TaskRequest(
    @SerializedName("title")
    val title: String,

    @SerializedName("description")
    val description: String? = null,

    @SerializedName("status")
    val status: TaskStatus? = null,

    @SerializedName("priority")
    val priority: TaskPriority? = null,

    @SerializedName("assignee_id")
    val assigneeId: Int? = null,

    @SerializedName("parent_id")
    val parentId: Int? = null,

    @SerializedName("project_id")
    val projectId: Int? = null,

    @SerializedName("due_date")
    val dueDate: String? = null,

    @SerializedName("estimated_minutes")
    val estimatedMinutes: Int? = null
)

File: ios-backup-20251014/AI-Proj-iOS/Models/Task.swift
//
//  Task.swift
//  AI-Proj iOS
//
//  任务相关数据模型
//

import Foundation

/// 任务模型
struct Task: Identifiable, Codable, Hashable {
    let id: Int
    let title: String
    let description: String?
    let status: TaskStatus
    let priority: TaskPriority
    let dueDate: Date?
    let createdAt: Date
    let updatedAt: Date
    let projectId: Int?
    let parentId: Int?

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case description
        case status
        case priority
        case dueDate = "due_date"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case projectId = "project_id"
        case parentId = "parent_id"
    }
}

/// 任务状态
enum TaskStatus: String, Codable, CaseIterable {
    case todo = "todo"
    case inProgress = "in_progress"
    case completed = "completed"
    case blocked = "blocked"
    case cancelled = "cancelled"

    var displayName: String {
        switch self {
        case .todo: return "待办"
        case .inProgress: return "进行中"
        case .completed: return "已完成"
        case .blocked: return "受阻"
        case .cancelled: return "已取消"
        }
    }
}

/// 任务优先级
enum TaskPriority: String, Codable, CaseIterable {
    case low = "low"
    case medium = "medium"
    case high = "high"
    case critical = "critical"

    var displayName: String {
        switch self {
        case .low: return "低"
        case .medium: return "中"
        case .high: return "高"
        case .critical: return "紧急"
        }
    }
}

/// 任务筛选器
enum TaskFilter: String, CaseIterable {
    case all = "all"
    case todo = "todo"
    case inProgress = "in_progress"
    case completed = "completed"

    var displayName: String {
        switch self {
        case .all: return "全部"
        case .todo: return "待办"
        case .inProgress: return "进行中"
        case .completed: return "已完成"
        }
    }
}
